ui/src/components/Settings/Notifications/index.tsx

```diff
@@ -0,0 +1,199 @@
+import { useContext, useEffect, useState } from 'react'
+import SettingsContext from '../../../contexts/settings-context'
+import React from 'react'
+import CreateNotificationModal, {
+  AgentConfiguration,
+} from './CreateNotificationModal'
+import GetApiHandler, {
+  DeleteApiHandler,
+  PostApiHandler,
+} from '../../../utils/ApiHandler'
+import Button from '../../Common/Button'
+import {
+  DocumentAddIcon,
+  PlusCircleIcon,
+  TrashIcon,
+} from '@heroicons/react/solid'
+import ExecuteButton from '../../Common/ExecuteButton'
+import { debounce } from 'lodash'
+import { useToasts } from 'react-toast-notifications'
+
+const NotificationSettings = () => {
+  const settingsCtx = useContext(SettingsContext)
+  const [addModalActive, setAddModalActive] = useState(false)
+  const [configurations, setConfigurations] = useState<AgentConfiguration[]>()
+  const [editConfig, setEditConfig] = useState<AgentConfiguration>()
+  const { addToast } = useToasts()
+
+  useEffect(() => {
+    document.title = 'Maintainerr - Settings - Notifications'
+    GetApiHandler('/notifications/configurations').then((configs) =>
+      setConfigurations(configs),
+    )
+  }, [])
+
+  useEffect(() => {
+    GetApiHandler('/notifications/configurations').then((configs) =>
+      setConfigurations(configs),
+    )
+  }, [addModalActive])
+
+  const doEdit = (id: number) => {
+    const config = configurations?.find((c) => c.id === id)
+
+    setEditConfig(config)
+    setAddModalActive(!addModalActive)
+  }
+
+  function confirmedDelete(id: any) {
+    DeleteApiHandler(`/notifications/configuration/${id}`).then(() => {
+      setConfigurations(configurations?.filter((c) => c.id !== id))
+    })
+  }
+
+  const doTest = () => {
+    PostApiHandler(`/notifications/test`, {}).then(() => {
+      addToast(
+        "Test notification deployed to all agents with the 'Test' type",
+        {
+          autoDismiss: true,
+          appearance: 'success',
+        },
+      )
+    })
+  }
+
+  return (
+    <div className="h-full w-full">
+      <div className="section-settings h-full w-full">
+        <h3 className="heading">Notification Settings</h3>
+        <p className="description">Notification configuration</p>
+      </div>
+      <div className="m-auto mb-5 flex">
+        <div className="ml-2 mr-auto sm:mr-0">
+          <ExecuteButton
```

Comment:
The first time around I found the test notification thing confusing. I added a notification, smashed Test Notifications a bunch of times and went on a entrypoint spree because nothing was happening. Only when I went back to the modal did I see the 'Test Notification' type and then it clicked in my head.

I think a friendlier version would be having the Test button within the notification modal that just uses the Test Notification type in the background. Looking at the code it might just be a case of passing the non-saved settings up, create a new agent (maybe can extract the factory logic from registerConfiguredAgents) and calling sendNotificationToAgent?

Reply:
There's also no feedback after testing currently, and for Gotify (what I've tested so far) the error & debug don't really help much (I'll add a separate comment for that).

Exceptions in the agents send() could bubble up (or throw a new one) with the actual exception message, that could be returned to the user when they hit the /test endpoint so they don't have to dig through logs.

Reply:
Good point, I hadn't really considered that. Looking at it now, it does seem odd. I took the easiest route at the time.
Moving the test button to the creation modal seems like the best approach, and we could also remove the 'test' type in that case.